File: src/doctor.rs
//! `focci doctor` — print detected configuration and integration status,
//! so the user can see what would be focused and whether the hooks are wired.

use std::env;

use crate::{focus, install};

fn or_unset(key: &str) -> String {
    env::var(key).unwrap_or_else(|_| "<unset>".to_string())
}

fn yes_no(present: bool) -> &'static str {
    if present {
        "yes"
    } else {
        "no"
    }
}

pub fn run() {
    println!("focci {}", env!("CARGO_PKG_VERSION"));
    println!();

    println!("Host app detection:");
    match focus::resolve_bundle() {
        Some(resolution) => {
            println!(
                "  bundle id            : {} (via {})",
                resolution.bundle_id, resolution.source
            )
        }
        None => println!("  bundle id            : <none> — set FOCCI_BUNDLE_ID to override"),
    }
    println!(
        "  __CFBundleIdentifier : {}",
        or_unset("__CFBundleIdentifier")
    );
    println!("  TERM_PROGRAM         : {}", or_unset("TERM_PROGRAM"));
    println!("  debounce             : {} ms", focus::debounce_ms());
    println!("  binary               : {}", install::binary_command());
    println!();

    let claude_path = install::claude_settings_path();
    let (stop, notification) = install::claude_status(&claude_path);
    println!("Claude Code ({}):", claude_path.display());
    println!("  Stop hook            : {}", yes_no(stop));
    println!("  Notification hook    : {}", yes_no(notification));
    println!();

    let codex_path = install::codex_config_path();
    println!("Codex ({}):", codex_path.display());
    println!(
        "  notify wired         : {}",
        yes_no(install::codex_status(&codex_path))
    );
    println!();

    println!("Tip: switch to another window, then run `focci focus` to test.");
}

File: cmd/doctor.go
package cmd

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"

	"github.com/HabibUllahKhanBarakzai/focci/internal/config"
	"github.com/HabibUllahKhanBarakzai/focci/internal/focus"
)

var doctorCmd = &cobra.Command{
	Use:   "doctor",
	Short: "Print detected configuration and integration status",
	Run: func(cmd *cobra.Command, args []string) {
		runDoctor()
	},
}

func orUnset(key string) string {
	if value, ok := os.LookupEnv(key); ok {
		return value
	}
	return "<unset>"
}

func yesNo(present bool) string {
	if present {
		return "yes"
	}
	return "no"
}

func runDoctor() {
	fmt.Printf("focci %s\n", version)
	fmt.Println()

	fmt.Println("Host app detection:")
	if resolution, ok := focus.ResolveBundle(); ok {
		fmt.Printf("  bundle id            : %s (via %s)\n", resolution.BundleID, resolution.Source)
	} else {
		fmt.Println("  bundle id            : <none> — set FOCCI_BUNDLE_ID to override")
	}
	fmt.Printf("  __CFBundleIdentifier : %s\n", orUnset("__CFBundleIdentifier"))
	fmt.Printf("  TERM_PROGRAM         : %s\n", orUnset("TERM_PROGRAM"))
	fmt.Printf("  debounce             : %d ms\n", focus.DebounceMS())
	fmt.Printf("  binary               : %s\n", config.BinaryCommand())
	fmt.Println()

	claudePath := config.ClaudeSettingsPath()
	stop, notification := config.ClaudeStatus(claudePath)
	fmt.Printf("Claude Code (%s):\n", claudePath)
	fmt.Printf("  Stop hook            : %s\n", yesNo(stop))
	fmt.Printf("  Notification hook    : %s\n", yesNo(notification))
	fmt.Println()

	codexPath := config.CodexConfigPath()
	fmt.Printf("Codex (%s):\n", codexPath)
	fmt.Printf("  notify wired         : %s\n", yesNo(config.CodexStatus(codexPath)))
	fmt.Println()

	fmt.Println("Tip: switch to another window, then run `focci focus` to test.")
}
